[OSDOCS#18230]: Support TNA with Platform=none (#110815)
openshift/openshift-docs: installing/installing_two_node_cluster/about-two-node-arbiter-installation.adoc

= Two-Node with Arbiter
:context: about-two-node-arbiter-installation

> [role="_abstract"]

Reviewer: We should not add this label just to make Vale be quiet. We should only add this label in front of a valid short description. This existing content is too long does not reflect user focus. Writing a proper short desc is out of scope for your work, but it does need to be changed, and if we throw a label on it it disappears from the to-do list. Given that this PR is not a CQA, I suggest deleting this label so that the content gets a good short description when CQA work is done on it.

```suggestion
```

More info about how to write a short description here for reference: https://docs.google.com/document/d/1xRc_licMTbLrLG9vmJ4bP9aPk6mzwwly5lGlkRz9d9s

Author: As of now, I simply deleted [role="_abstract"]. May be when 4.22 feature work is in much control, we can consider implementing CQA activities for TNA and TNF.

Reviewer: I think that makes the most sense as well.

An {product-title} cluster with two control plane nodes and one local arbiter node is a compact, cost-effective {product-title} topology. The arbiter node stores the full etcd data, maintaining an etcd quorum and preventing split brain. The arbiter node does not run the additional control plane components `kube-apiserver` and `kube-controller-manager`, nor does it run workloads.

To install a cluster with two control plane nodes and one local arbiter node, assign an arbiter role to at least one of the nodes and set the control plane node count for the cluster to 2. Although {product-title} does not currently impose a limit on the number of arbiter nodes, the typical deployment includes only one to minimize the use of hardware resources.

After installation, you can add additional arbiter nodes to a cluster with two control plane nodes and one local arbiter node but not to a standard multi-node cluster. It is also not possible to convert between a two-node OpenShift cluster with a local node arbiter and standard topology.

[NOTE]
====

> For a cluster with an arbiter, the same networking requirements as a regular cluster for connectivity between machines apply. See xref:../../installing/installing_platform_agnostic/installing-platform-agnostic.adoc#installation-network-connectivity-user-infra_installing-platform-agnostic[Network connectivity requirements]

Reviewer: The Vale rule here is correct, we cannot have in-line links in assemblies.
```suggestion
For a cluster with an arbiter, the same networking requirements as a regular cluster for connectivity between machines apply. For more information, see _Network connectivity requirements_.
```

Add the link to the _Additional resources_ section instead.

====

> You can install a cluster with two control plane nodes and one local arbiter node by using one of the following methods:

Reviewer: This is an issue in the existing content, we can't have a sentence leading into the _Additional resources_ section like this. I would suggest:
```suggestion
You can install a cluster with two control plane nodes and one local arbiter node on installer-provisioned infrastructure or on user-provisioned infrastructure. You can also by use the Agent-based Installer.
```

Alternatively, you could just delete this sentence entirely. It doesn't really add much since the list is fairly self explanatory.

[role="_additional-resources"]
.Additional resources

> * xref:../installing_bare_metal/ipi/ipi-install-installation-workflow.adoc#ipi-install-config-local-arbiter-node_ipi-install-installation-workflow[Configuring a local arbiter node with Installer-provisioned infrastructure]

Reviewer: ```suggestion
* xref:../installing_bare_metal/ipi/ipi-install-installation-workflow.adoc#ipi-install-config-local-arbiter-node_ipi-install-installation-workflow[Configuring a local arbiter node with installer-provisioned infrastructure]
```

* xref:../../installing/installing_with_agent_based_installer/installing-with-agent-based-installer.adoc#installing-ocp-agent-local-arbiter-node_installing-with-agent-based-installer[Configuring a local arbiter node with the Agent-based Installer]

> * xref:../installing_bare_metal/upi/installing-bare-metal.adoc#upi-install-config-local-arbiter-node_installing-bare-metal[Configuring a local arbiter node with User-provisioned infrastructure]

Reviewer: ```suggestion
* xref:../installing_bare_metal/upi/installing-bare-metal.adoc#upi-install-config-local-arbiter-node_installing-bare-metal[Configuring a local arbiter node with user-provisioned infrastructure]
```
